yarn-project/prover-client/src/orchestrator/orchestrator.ts

```diff
@@ -394,118 +478,83 @@ export class ProvingOrchestrator implements BlockProver {
-      const header = await this.extractBlockHeader();
-
-      logger?.debug(`Updating and validating root trees`);
-      await this.db.updateArchive(header);
-
-      await validateBlockRootOutput(rootRollupOutputs, header, this.db);
-
-      // Collect all new nullifiers, commitments, and contracts from all txs in this block
-      const gasFees = this.provingState.globalVariables.gasFees;
-      const nonEmptyTxEffects: TxEffect[] = this.provingState!.allTxs.map(txProvingState =>
-        toTxEffect(txProvingState.processedTx, gasFees),
-      ).filter(txEffect => !txEffect.isEmpty());
-      const blockBody = new Body(nonEmptyTxEffects);
-
-      const l2Block = L2Block.fromFields({
-        archive: rootRollupOutputs.newArchive,
-        header: header,
-        body: blockBody,
-      });
+      const block = this.provingState?.blocks[index ?? this.provingState?.blocks.length - 1];
 
-      if (!l2Block.body.getTxsEffectsHash().equals(header.contentCommitment.txsEffectsHash)) {
-        logger.debug(inspect(blockBody));
-        throw new Error(
-          `Txs effects hash mismatch, ${l2Block.body
-            .getTxsEffectsHash()
-            .toString('hex')} == ${header.contentCommitment.txsEffectsHash.toString('hex')} `,
-        );
+      if (!block || !block.blockRootRollupPublicInputs || !block.finalProof || !block.block) {
+        throw new Error(`Invalid proving state, a block must be proven before it can be finalised`);
       }
 
-      logger.info(`Orchestrator finalised block ${l2Block.number}`);
-
-      this.provingState.block = l2Block;
-
       const blockResult: ProvingBlockResult = {
-        proof: this.provingState.finalProof,
-        aggregationObject: this.provingState.finalProof.extractAggregationObject(),
-        block: l2Block,
+        proof: block.finalProof,
+        aggregationObject: block.finalProof.extractAggregationObject(),
+        block: block.block!,
       };
 
       pushTestData('blockResults', {
         proverId: this.proverId.toString(),
         vkTreeRoot: getVKTreeRoot().toString(),
-        block: l2Block.toString(),
-        proof: this.provingState.finalProof.toString(),
+        block: blockResult.block.toString(),
+        proof: blockResult.proof.toString(),
         aggregationObject: blockResult.aggregationObject.map(x => x.toString()),
       });
 
-      return blockResult;
+      return Promise.resolve(blockResult);
     } catch (err) {
       throw new BlockProofError(
         err && typeof err === 'object' && 'message' in err ? String(err.message) : String(err),
-        this.provingState?.allTxs.map(x => Tx.getHash(x.processedTx)) ?? [],
+        this.provingState?.blocks[index ?? this.provingState?.blocks.length - 1]?.allTxs.map(x =>
+          Tx.getHash(x.processedTx),
+        ) ?? [],
       );
     }
   }
 
+  /**
+   * Returns the proof for the current epoch.
+   * Requires proving to have been completed.
+   */
+  public finaliseEpoch() {
```

Comment:
We can probably get rid of this now. Previously the prover called this to insert the archive but this is done after each block. So presumably now we can just resolve the `ProvingTicket` promise and everything is done, which is a much nider interface?

Reply:
Absolutely. I wanted to keep existing interfaces (and even proofs for blocks) until the epoch flow was working, but last thing to do will be to clean up interfaces and delete the block proof flow.